src/language-js/utils.js

```diff
@@ -464,6 +464,7 @@ const simpleTypeAnnotations = new Set([
   "BigIntLiteralTypeAnnotation",
   "NumberLiteralTypeAnnotation",
   "TSLiteralType",
+  "TSTemplateLiteralType",
```

Comment:
Not sure about this. Unlike other node types in this list, template literal types aren't that simple and can contain complex expressions. We can keep them here though and see if users will complain about it. Probably it's okay.

Reply:
That's also what I thought 😄